### DISCUSSION TITLE
[useController]: Type error when trying to access `fieldState.error.message`

### DISCUSSION THREAD

**Discussion by manskron** [react-hook-form/react-hook-form#6708] [Q&A]
### Version Number

7.17.0

### Codesandbox/Expo snack

https://codesandbox.io/s/heuristic-stonebraker-b514g?file=/src/Input.tsx

### Steps to reproduce

**Describe the bug**
When using `useController`, trying to access `fieldState.error.message` results in a type error.

**To reproduce**
1. Go to 'https://codesandbox.io/s/heuristic-stonebraker-b514g?file=/src/Input.tsx'
2. Look at the type error on line 27

### Expected behaviour

No type errors.

### Relevant log output

```shell
TS2339: Property 'message' does not exist on type 'NonNullable>, PathValue ...>>>>'
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
you can try with the new type which just got released, https://github.com/react-hook-form/react-hook-form/releases/tag/v7.17.0

**Reply by jgradzki** [reply]
@bluebill1049 Hi, I have same problem. I see that you have reverted "FieldPathWithValue" and made some changes in other types, but this is still a problem for me:

```
export type FormControllerProps<
  TFieldValues extends FieldValues = FieldValues,
  TName extends Path<TFieldValues> = Path<TFieldValues>
> = ControllerProps<TFieldValues, TName> & FormControlProps;

const FormController = <
  TFieldValues extends FieldValues = FieldValues,
  TName extends Path<TFieldValues> = Path<TFieldValues>
>({
  render,
  ...props
}: FormControllerProps<TFieldValues, TName>): JSX.Element => {
  return (
    <Controller
      {...props}
      render={(data) => (
        <FormControl
          id={props.name}
          error={data.fieldState.error && data.fieldState.error.message}
          {...props}
        >
          {render(data)}
        </FormControl>
      )}
    />
  );
};
```

```
 Property 'message' does not exist on type 'DeepMapImpl<DeepPartialImpl<UnionLike<PathValue<TFieldValues, TName>>>, FieldError>'.  TS2339
```

**Reply by jgradzki** [reply]
workaround is to cast type:

(data.fieldState.error as FieldError).message

**Reply by davbrito** [reply]
I hope this could ve solved and we don't need a workaround. This error is somewhat annoying me.

**Comment by bluebill1049** [comment]
also in your example, you are missing the extend

```tsx
import { useController, UseControllerProps } from "react-hook-form";
import { FieldValues } from 'react-hook-form';
interface IProps<FormValues> extends UseControllerProps<FormValues> {
  type: string;
}

export const Input = <FormValues extends FieldValues>({
  name,
  control,
  rules,
  type,
}: IProps<FormValues>) => {
  const {
    field: { ref, ...inputProps },
    fieldState: { error }
  } = useController({
    name,
    control,
    rules
  });

  console.log(error);

  return (
    <div>
      <input ref={ref} type={type} {...inputProps}/>
      <p>{error?.message}</p>
    </div>
  );
};
```

**Reply by ghost** [reply]
Hey @bluebill1049. 
Just want to start out telling you im a big fan of your library and appreciate all the work that goes into it.

I too am experiencing problems with the message property on the `fieldState.error` object. Am I by any chance missing something obvious?

```
interface SimpleSelectControlProps<T> extends UseControllerProps<T> {
  options: string[];
  placeholder?: string;
}

export const SimpleSelectControl = <T extends FieldValues>(
  props: SimpleSelectControlProps<T>
) => {
  const { options, placeholder, name, rules, control } = props;

  const {
    field: { value, onChange },
    fieldState: { error },
  } = useController({
    name,
    control,
    rules,
  });

  return (
    <Column>
      <SimpleSelect
        onChange={onChange}
        value={value}
        options={options}
        placeholder={placeholder}
      />
      {error != null ? (
        <ErrorText>{error?.message}</ErrorText>  // <-- error 
      ) : null}
    </Column>
  );
};
```

**Reply by davbrito** [reply]
I'm having exactly the same issue with a component like yours.

**Reply by ghost** [reply]
@davbrito Im experiencing this issue on v. 7.17.1
Downgrading to v. 7.14.0 will fix it.

**Reply by bluebill1049** [reply]
we are try to fix that issue, will patch that in the next release.
